src/cryptonote_basic/cryptonote_basic.h

```diff
@@ -480,13 +546,50 @@ void serialize_value(Archive& ar, block& b) {
         throw oxen::traced<std::invalid_argument>{"too many txs in block"};
     if (b.major_version >= hf::hf16_pulse)
         field(ar, "signatures", b.signatures);
-    if (b.major_version >= hf::hf19_reward_batching) {
-        field_varint(ar, "height", b.height);
-        field(ar, "service_node_winner_key", b.service_node_winner_key);
-        field(ar, "reward", b.reward);
+    if (b.major_version == hf::hf19_reward_batching) {
+        // Oxen 10.x included three fields here, height, service_node_winner_key (which was actually
+        // the pulse block producer, not necessarily the winner), and reward.  None are actually
+        // needed (because we can always compute the values they must have), and only reward is used
+        // as of Oxen 11.x (because it's very useful to keep the reward stored alongside the block).
+        //
+        // Dropping them is a bit tricky, though, because Oxen 10.x nodes still expect/use those
+        // values so we have to keep sending them as long as there are Oxen 10 nodes on the network.
+        // We build a bit of forward compatibility in here, however, by accepting *either*:
+        //
+        // - a single 0 varint value; this is only sent by a future Oxen 11.x release that has
+        //   dropped Oxen 10 interoperability.  In this case we just set the height and producer
+        //   fields to 0/null.  (Thus 11.1.x nodes with compatibility in place will still be able to
+        //   sync from 11.2.x that stop sending these fields).
+        // - Otherwise it's a height and we *also* read the Oxen 10-compatible height/producer
+        //   values immediately after it.
+        //
+        // TODO: come back after all nodes are across the Oxen 11 fork and remove the
+        // interoperability *and* the oxen10_pulse_provider field and leave just the dummy 0 value
+        // height encoding (currently disabled in the `else` below).
+        if constexpr (SUPPORT_OXEN10_INTEROP) {
+            field_varint(ar, "height", b._height);
+            if (Archive::is_deserializer && b._height == 0) {
+                // Incoming from the future where Oxen 11 doesn't use compat mode any more
+                b.oxen10_pulse_producer = crypto::null<crypto::public_key>;
+                b.reward = 0;
+            } else {
+                // Message to/from someone in Oxen 10 compat mode
+                field(ar, "pulse_producer", b.oxen10_pulse_producer);
+                field(ar, "reward", b.reward);
+            }
+        } else {
+            // TODO: enable this code once we only care about supporting Oxen 11+ nodes.
+
+            // Serialize a 0 height, even if the value we have isn't 0 (which it might not be if we
+            // added this block to our DB while running Oxen 10).
+            [[maybe_unused]] uint64_t no_height = 0;
+            field_varint(ar, "height", no_height);
```

Comment:
When someone on Oxen 11+ syncs the chain and the SUPPORT_OXEN10_INTEROP branch is disabled/deleted, blocks from HF19 are going to be missing the rewards value?

The `pulse_producer` is fine because we pull out the value from `tx_extra`. I think it's even possible to get rid of it now in the `SUPPORT_OXEN10_INTEROP` branch though I'm guessing that's probably going to have a cyclical include problem.

```cpp
if constexpr (SUPPORT_OXEN10_INTEROP) {
    // ...
    if (Archive::is_deserializer && b._height == 0) {
        // ...
    } else {
        auto pulse_producer = cryptonote::get_service_node_winner_from_tx_extra(b.miner_tx->extra);
        field(ar, "pulse_producer", pulse_producer);
        // ...
    }
}
```

But when a Oxen 11+ node syncs blocks from HF19 from an Oxen 11+ node it's going to set `rewards` to 0, then for example wouldn't this branch fail in `Blockchain::validate_miner_transaction`?

```cpp
  } else if (version == hf::hf19_reward_batching) {
      if (b.reward != reward_parts.miner_fee + reward_parts.service_node_total) {
          log::error(
                  log::Cat("verify"),
                  "block reward for batching is incorrect: {} != {} ({} SN + {} tx fees)",
                  print_money(b.reward),
                  print_money(reward_parts.miner_fee + reward_parts.service_node_total),
                  print_money(reward_parts.service_node_total),
                  print_money(reward_parts.miner_fee));
          return false;
      }
}
```

Reply:
> pulse_producer on the otherhand is fine because we pull out the value from tx_extra. I think it's even possible to get rid of it now in the SUPPORT_OXEN10_INTEROP branch though I'm guessing that's probably going to have a cyclical include problem.

I looked at this, and I don't think it is because current Oxen 10 code uses that value from the serialized block in some places -- so dropping it now would break Oxen 10's ability to sync from an Oxen 11 node in the period where we have both on the network during the last days of HF19.

Reply:
> > pulse_producer on the otherhand is fine because we pull out the value from tx_extra. I think it's even possible to get rid of it now in the SUPPORT_OXEN10_INTEROP branch though I'm guessing that's probably going to have a cyclical include problem.
> 
> I looked at this, and I don't think it is because current Oxen 10 code uses that value from the serialized block in some places -- so dropping it now would break Oxen 10's ability to sync from an Oxen 11 node in the period where we have both on the network during the last days of HF19.

In this case it wouldn't be dropping it because in the serialisation phase an Oxen 11 node can retrieve the pulse producer value from a HF19 block (by extracting it from the TX extra) and serialise it so that it's always present on the wire when its being synced for Oxen 10 nodes. 

Though it's pretty minor .. because we can just wait until we drop the code when we're all on Oxen 11.